mozdownload/parser.py

```diff
@@ -13,8 +13,8 @@
 class DirectoryParser(HTMLParser):
     """Class to parse directory listings"""
 
-    def __init__(self, url, authentication=None, timeout=None):
-        self.authentication = authentication
+    def __init__(self, url, session=None, timeout=None):
+        self.session = session or requests.Session()
```

Comment:
Indeed, if we pass None (the default value) a session will be automatically created without authentication - still it is of course possible to pass a session that owns authentication data and this is what the scrappers are doing. I wanted to keep the argument optional, in case it was used by some test (yeah, I'm lazy;). If you don't want this behavior, I can force the session to be given (maybe as the first argument).